internal/validators/os/os_valdiator_test.go

```diff
@@ -0,0 +1,114 @@
+package os
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/go-logr/logr"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/validator-labs/validator-plugin-maas/api/v1alpha1"
+)
+
+type DummyMaaSAPIClient struct {
+	images []entity.BootResource
+}
+
+func (d *DummyMaaSAPIClient) ListOSImages() ([]entity.BootResource, error) {
+	return d.images, nil
+}
+
+func TestFindingBootResources(t *testing.T) {
+
+	type TestCase struct {
+		Name        string
+		ruleService *ImageRulesService
+		imageRules  []v1alpha1.OSImageRule
+		errors      []error
+		details     []string
+	}
+
+	testCases := []TestCase{
+		{
+			Name: "image is found in MaaS",
+			ruleService: NewImageRulesService(
+				logr.Logger{},
+				&DummyMaaSAPIClient{
+					images: []entity.BootResource{
+						{
+							Name:         "Ubuntu",
+							Architecture: "amd64/ga-20.04",
+						},
+					},
+				},
+			),
+			imageRules: []v1alpha1.OSImageRule{
+				{Name: "Image found", OSImages: []v1alpha1.OSImage{
+					{OSName: "Ubuntu", Architecture: "amd64/ga-20.04"},
+				}},
+			},
+			errors:  make([]error, 0),
```

Comment:
The zero value of a slice is `nil`, so typically you'd set `errors: nil` in the case that no errors are expected